### PR TITLE
[chip,dv] add local alert event to  rstmgr_alert_info test

### PR DESCRIPTION
**Pull request by jdonjdon**
This PR updates 'rstmgr_alert_info_test' to address (#14140)
by adding loc_alert_cause field in aleart handler crash dump. 
The loc_alert_cause field is set by alert ping timeout event.

Signed-off-by: Jaedon Kim <jdonjdon@google.com>

### CODE REVIEW

File: sw/device/tests/rstmgr_alert_info_test.c [hunk 1]
```diff
@@ -63,6 +64,11 @@
     reset.
   - Read alert_info from rstmgr and compare with expected value.
 
+  Round 4: Local alert
+  - Trigger local alert by setting ping timeout value to 1.
+  - Once alert triggers interrupt, call sw_device reset from interrupt
+    handler
```
Comment (cindychip) [inline sw/device/tests/rstmgr_alert_info_test.c:70]:
```suggestion
    handler.
```

File: sw/device/tests/rstmgr_alert_info_test.c [hunk 2]
```diff
@@ -557,12 +638,22 @@ static void collect_alert_dump_and_compare(test_round_t round) {
...
-  print_alert_cause(kExpectedInfo[round]);
-  for (int i = 0; i < ALERT_CAUSE_CNT; ++i) {
-    CHECK(kExpectedInfo[round].alert_cause[i] == actual_info.alert_cause[i]);
+  if (round == kRound4) {
+    // Check local alert only.
+    // While testing ping timeout for local alert,
+    // global alert ping timeout can be trigerred
+    // dut to short timeout value.
+    // However, alert source of this ping timeout can be choose randomly,
+    // we cannot predic exact source of remote alert.
```
Comment (cindychip) [inline sw/device/tests/rstmgr_alert_info_test.c:647]:
```suggestion
    // we cannot predict exact source of remote alert.
```

File: sw/device/tests/rstmgr_alert_info_test.c [hunk 3]
```diff
@@ -557,12 +638,22 @@ static void collect_alert_dump_and_compare(test_round_t round) {
...
-  LOG_INFO("expected alert cause:");
-  print_alert_cause(kExpectedInfo[round]);
-  for (int i = 0; i < ALERT_CAUSE_CNT; ++i) {
-    CHECK(kExpectedInfo[round].alert_cause[i] == actual_info.alert_cause[i]);
+  if (round == kRound4) {
+    // Check local alert only.
+    // While testing ping timeout for local alert,
+    // global alert ping timeout can be trigerred
+    // dut to short timeout value.
+    // However, alert source of this ping timeout can be choose randomly,
```
Comment (cindychip) [inline sw/device/tests/rstmgr_alert_info_test.c:646]:
```suggestion
    // However, alert source of this ping timeout can be chosen randomly,
```

File: sw/device/tests/rstmgr_alert_info_test.c (hunk 2 above)
Comment (cindychip) [reply sw/device/tests/rstmgr_alert_info_test.c:647]: Actually, I wonder if you mean this;

```
// Alert ping timeout might randomly trigger a spurious alert, as documented in issue #2321
// So here we only check the local alert cause, but not the alert_cause register.
```

**Review by cindychip — APPROVED** [review]

File: sw/device/tests/rstmgr_alert_info_test.c (hunk 1 above)
Comment (jdonjdon) [reply sw/device/tests/rstmgr_alert_info_test.c:70]: fixed

File: sw/device/tests/rstmgr_alert_info_test.c (hunk 2 above)
Comment (jdonjdon) [reply sw/device/tests/rstmgr_alert_info_test.c:647]: Yes. Let me update this clause.

File: sw/device/tests/rstmgr_alert_info_test.c (hunk 3 above)
Comment (jdonjdon) [reply sw/device/tests/rstmgr_alert_info_test.c:646]: fixed

**Pull request merged**